src/chainparams.cpp

```diff
@@ -280,8 +280,10 @@ class SigNetParams : public CChainParams {
 
         if (!args.IsArgSet("-signetchallenge")) {
             bin = ParseHex("512103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be430210359ef5021964fe22d6f8e05b2463c9540ce96883fe3b278760f048f5189f2e6c452ae");
+            vSeeds.emplace_back("seed.signet.bitcoin.sprovoost.nl");
+
+            // TODO: remove hardcoded nodes once there are more DNS seeds
```

Comment:
I am generally not a fan of `TODO` in the source code, especially ones that are impossible to solve by writing code. This one require someone to run a signet dns seed node. There is no risk in keeping the hardcoded addresses, so the TODO can simply be removed. If you think it should be kept, a simple `// Hardcoded addresses can be removed once there are more DNS seeds` would achieve the same without any downside. Currently this is just another TODO that wastes (new) developers time that are looking for stuff that needs to be fixed.

Even if this was something that can be fixed by writing code, I think we had a discussion to prefer GitHub (good first) issues over that.

Reply:
> There is no risk in keeping the hardcoded addresses, so the TODO can simply be removed.

Actually there is a risk: the hardcoded nodes are worse in terms of privacy. Although this is not a big deal with Signet. Removing the word "TODO" is fine with me though.

Reply:
DNS seeds are bad in terms of security: https://bitcoin.stackexchange.com/questions/109852/dns-seeds-and-security

Reply:
The hardcoded fallbacks aren't used when your `peers.dat` already contains peer's addresses.

If the fallbacks are compromised, it will be bad for privacy/security regardless of whether they are retrieved from a hardcoded DNS or directly a hardcoded address.

Reply:
@prayank23: Blanket statements like "X is bad for security" and "Y is bad for privacy" isn't particularly helpful when you don't provide a comparison or discuss the trade-offs. Would you rather use hardcoded IP addresses than DNS seeds? Or would you rather a new node fails to find any peers rather than using hardcoded IP addresses or DNS seeds? The eclipse attack only works if the attacker controls all your peers. As long as you have one honest peer then you will receive valid blocks with sufficient proof of work that your attacker is unable to replicate. Hence using multiple DNS seeds in mainnet's case means the attacker would have to hack multiple domains. Not impossible but nothing is 100 percent secure. You just provide the best solution that is available. In signet's case we need more DNS seeds.

Reply:
@michaelfolkson 

> Blanket statements like "X is bad for security" and "Y is bad for privacy" isn't particularly helpful when you don't provide a comparison or discuss the trade-offs. Would you rather use hardcoded IP addresses than DNS seeds?

Its not a blanket statement. Has some background and I was assuming people involved in this discussion know what I am talking about.

> Would you rather use hardcoded IP addresses than DNS seeds?

I will prefer using IP for 9 DNS seeds used by Bitcoin Core and reasons were discussed or obvious in IRC few days back when we discussed with Sipa and Dhruv. Also change how things work and redirection remains for all.

FYI: Indian government loves to ban things based on domains so web telegram didn't work here for lot of ISPs but client worked because it used IPs and not domains. Other reasons for my comment are mentioned. If you need details on privacy thing: https://github.com/bitcoin/bitcoin/pull/22317

Reply:
> I will prefer using IP for 9 DNS seeds used by Bitcoin Core

So you are arguing to ditch DNS seeds entirely for mainnet in favor of hardcoded IP addresses? Or using a mixture of both DNS seeds and hardcoded IP addresses? The Signet approach is just copying the mainnet approach just with fewer DNS seeds.

Reply:
First let me clarify:

My comment here was a response to Sjors and nothing else. I don't care what signet uses. Also I agree with comment by Marcofalke. It makes sense.

> So you are arguing to ditch DNS seeds entirely for mainnet in favor of hardcoded IP addresses? Or using a mixture of both DNS seeds and hardcoded IP addresses?

I am saying lets use same process as we do for Tor and 9 seeds have IP address that responds with p2p message and lot of addresses in it. I have to trust 9 people, why not use IP for them?